lib/utils.cjs

```diff
@@ -51,7 +51,7 @@ module.exports = {
 
       matches.forEach(function (file) {
```

Comment:
With `directory/**` glob pattern, it reads **every** package.json and will use any `package.json` file, including those in `node_modules`. And of course it will use the last one, from dependencies.

Reply:
This PR changed from matching anything ending in `package.json` to anything named `package.json`.
Therefore the number of matches should be reduced.

Globbing was present before.